experimental/dds/tree2/src/core/rebase/utils.ts

```diff
@@ -76,53 +85,90 @@ export function rebaseBranch<TChange>(
 export function rebaseBranch<TChange>(
 	changeRebaser: ChangeRebaser<TChange>,
 	sourceHead: GraphCommit<TChange>,
-	newBase: GraphCommit<TChange>,
+	targetCommit: GraphCommit<TChange>,
 	targetHead: GraphCommit<TChange>,
-): [newSourceHead: GraphCommit<TChange>, sourceChange: TChange];
+): [
+	newSourceHead: GraphCommit<TChange>,
+	sourceChange: TChange | undefined,
+	commits: {
+		newBase: GraphCommit<TChange>;
+		deletedSourceCommits: GraphCommit<TChange>[];
+		newSourceCommits: GraphCommit<TChange>[];
+	},
+];
 export function rebaseBranch<TChange>(
 	changeRebaser: ChangeRebaser<TChange>,
 	sourceHead: GraphCommit<TChange>,
-	newBase: GraphCommit<TChange>,
-	targetHead = newBase,
-): [newSourceHead: GraphCommit<TChange>, sourceChange: TChange] {
+	targetCommit: GraphCommit<TChange>,
+	targetHead = targetCommit,
+): [
+	newSourceHead: GraphCommit<TChange>,
+	sourceChange: TChange | undefined,
+	commits: {
+		newBase: GraphCommit<TChange>;
```

Comment:
What is the difference between newBase and newSourceHead?

Reply:
`newSourceHead` is the new head of the branch that got rebased, and `newBase` is the new location where it branches off of the target branch. I updated the `newBase` docs to make this clear.